api/base/views.py

```diff
@@ -127,9 +127,6 @@ def partial(item):
                 with transaction.atomic():
                     ret = view.handle_exception(e).data
 
-            # Allow request to be gc'd
-            ser._context = None
-
```

Comment:
Noticed immediate issues with trying to do `always_embed` for subject parents. 
It seemed if your page had more than one level of embeds, issues would be triggered, I believe because context was being overwritten here.   I do not understand the implications of removing this - was added in this PR https://github.com/CenterForOpenScience/osf.io/pull/5277/


One thing to note, `always_embed` seems to embed all the way up the hierarchy until there is nothing left to embed.

Reply:
I'm pretty sure this change is fine.

Some git archaeology:
The update in #5277 was using a global weakref dict (`CACHE`) to store serializers for embedded things. Clearing the serializer's context here was meant to break the reference to the current request, so the request (and the cached serializer) could be garbage collected. As shown by [this later commit](https://github.com/CenterForOpenScience/osf.io/commit/3fafa239a2331aed41b88eb71c741dc87b7218cb), that wasn't enough (I guess the cached serializer held another reference to the request?) and the global cache was replaced with a cache stored on the request itself. With that, the entire cache is discarded with the request and clearing the context here has no real benefit. Also, when a serializer is reused, [its context is reset anyway](https://github.com/pattisdr/osf.io/blob/9060f8de584fcf36c3d6b1ca5cc13cb0273137f0/api/base/views.py#L111).

Reply:
wow, thanks for digging into this!